src/padding.rs

```diff
@@ -1,12 +1,166 @@
-// Applies new padding to a BSON packet.
-// Returns the padded packet.
-fn encapsulate(pkt: &[u8]) -> Vec<u8> {
-    unimplemented!()
+use rand::{distributions::Uniform, CryptoRng, Rng, SeedableRng};
+use std::io::{Error, ErrorKind};
+
+/// Generates a random number between 2 and 32 (inclusive) with the given random number generator.
+///
+/// # Arguments
+///
+/// * `rng` - A mutable reference to a random number generator.
+///
+/// # Returns
+///
+/// A random u32 between 2 and 32 (inclusive).
+
+fn generate_random_num<R: Rng + CryptoRng>(rng: &mut R) -> u32 {
```

Comment:
This function could be inlined, and we don't need to use Distribution - there's simpler functions we can use for pulling randomness. See the AES example in the repo for how to do that.